File: soal9.py
def climbStairs(n):
    if n == 1:
        return 1
    if n == 2:
        return 2
    
    first, second = 1, 2
    for i in range(3, n + 1):
        first, second = second, first + second
    return second

# Contoh penggunaan
if __name__ == "__main__":
    n = int(input("Masukkan jumlah langkah: ").strip())
    print(climbStairs(n))

File: soal9.ts
function climbStairs(n: number): number {
    if (n === 1) return 1;
    if (n === 2) return 2;

    let first: number = 1;
    let second: number = 2;

    for (let i = 3; i <= n; i++) {
        let temp: number = first + second;
        first = second;
        second = temp;
    }

    return second;
}

// Contoh penggunaan
const n: number = parseInt(prompt("Masukkan jumlah langkah: ") || "0");
console.log(climbStairs(n));
